### PR TITLE
 add recipient flow

### PR DESCRIPTION
**Pull request by raincoatforever**
@shahmharsh

### CODE REVIEW

File: app/src/main/java/co/minesweepers/birthday/model/Memory.java [hunk 1]
```diff
@@ -61,6 +65,27 @@ public String getId() {
         return mId;
     }
 
+    public void setRecipient(final String name, Uri imageUri, final Utils.GenericOperationListener listener) {
```
Comment (shahmharsh) [inline app/src/main/java/co/minesweepers/birthday/model/Memory.java]: add @NonNull annotation

File: app/src/main/java/co/minesweepers/birthday/activities/GetRecipientActivity.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    mNameTxt.setClickable(false);
+                }
+                break;
+
+            case R.id.get_recipient_upload_pic:
+                showPictureChooser();
+                break;
+
+            case R.id.button_recipient_done:
+                Intent viewMemoryIntent = new Intent(this, AddPeopleActivity.class);
```
Comment (shahmharsh) [inline app/src/main/java/co/minesweepers/birthday/activities/GetRecipientActivity.java:72]: shouldnt you be setting the recipient in the memory before going to AddPeopleActivity?

Comment (raincoatforever) [reply app/src/main/java/co/minesweepers/birthday/activities/GetRecipientActivity.java:72]: look at line 106 below . Continue button is not visible to the user till mMemory.setRecipient triggers onSuccess callback . User won't see this button ( and won't go to AddPeopleActivity ) without setting a recipient .

File: app/src/main/java/co/minesweepers/birthday/model/Memory.java (hunk 1 above)
Comment (raincoatforever) [reply app/src/main/java/co/minesweepers/birthday/model/Memory.java]: done .

**Comment by shahmharsh** [comment]
👍

**Pull request merged**